sklearn/linear_model/tests/test_quantile.py

```diff
@@ -0,0 +1,126 @@
+# Authors: David Dale dale.david@mail.ru
+# License: BSD 3 clause
+
+import numpy as np
+import pytest
+import scipy.stats
+
+from sklearn.utils import check_random_state
+from sklearn.utils._testing import assert_allclose, assert_raises
+from sklearn.datasets import make_regression
+from sklearn.linear_model import HuberRegressor, QuantileRegressor
+from sklearn.utils.fixes import sp_version, parse_version
+
+_SCIPY_TOO_OLD = "requires at least scipy 1.0.0"
+
+
+@pytest.mark.parametrize(
+    'quantile, alpha, intercept, coef',
+    [
+        # for 50% quantile w/o regularization, any slope in [1, 10] is okay
+        [0.5, 0, 1, None],
+        # if positive error costs more, the slope is maximal
+        [0.51, 0, 1, 10],
+        # if negative error costs more, the slope is minimal
+        [0.49, 0, 1, 1],
+        # for a small lasso penalty, the slope is also minimal
+        [0.5, 0.01, 1, 1],
+        # for a large lasso penalty, the model predicts the constant median
+        [0.5, 100, 2, 0],
+    ]
+)
+@pytest.mark.skipif(sp_version < parse_version("1.0.0"), reason=_SCIPY_TOO_OLD)
+def test_quantile_toy_example(quantile, alpha, intercept, coef):
+    # test how different parameters affect a small intuitive example
+    X = [[0], [1], [1]]
+    y = [1, 2, 11]
+    model = QuantileRegressor(quantile=quantile, alpha=alpha).fit(X, y)
+    assert_allclose(model.intercept_, intercept, atol=1e-2)
+    if coef is not None:
+        assert_allclose(model.coef_[0], coef, atol=1e-2)
+    if alpha < 100:
+        assert model.coef_[0] >= 1
+    assert model.coef_[0] <= 10
+
+
+@pytest.mark.skipif(sp_version < parse_version("1.0.0"), reason=_SCIPY_TOO_OLD)
+def test_quantile_equals_huber_for_low_epsilon():
+    X, y = make_regression(n_samples=100, n_features=20, random_state=0,
+                           noise=1.0)
+    huber = HuberRegressor(epsilon=1+1e-4, alpha=1e-4).fit(X, y)
+    quant = QuantileRegressor(alpha=1e-4).fit(X, y)
+    assert_allclose(huber.intercept_, quant.intercept_, atol=1e-1)
+    assert_allclose(huber.coef_, quant.coef_, atol=1e-1)
+
+
+@pytest.mark.skipif(sp_version < parse_version("1.0.0"), reason=_SCIPY_TOO_OLD)
+def test_quantile_estimates_fraction():
+    # Test that model estimates percentage of points below the prediction
+    X, y = make_regression(n_samples=1000, n_features=20, random_state=0,
+                           noise=1.0)
+    for q in [0.5, 0.9, 0.05]:
+        quant = QuantileRegressor(quantile=q, alpha=0).fit(X, y)
+        fraction_below = np.mean(y < quant.predict(X))
+        assert_allclose(fraction_below, q, atol=1e-2)
+
+
+@pytest.mark.skipif(sp_version < parse_version("1.0.0"), reason=_SCIPY_TOO_OLD)
+def test_quantile_without_intercept():
```

Comment:
Could this test be incorporated in `test_quantile_equals_huber_for_low_epsilon` with the help of `@pytest.mark.parametrize('fit_intercept', [True, False])`?